protected/humhub/modules/file/models/File.php

```diff
@@ -112,12 +129,25 @@ public function behaviors()
     }
 
     /**
-     * Gets query for [[FileHistory]].
+     * @return array
+     * @noinspection PhpMissingReturnTypeInspection
+     * @noinspection ReturnTypeCanBeDeclaredInspection
+     */
+    public function transactions()
```

Comment:
Not sure we should implement this, at the current stage.

Reply:
Sure. I've added them in my PR #6408 back in June ... 😉

I have removed them. Although, I have added them as the update and delete actions have some relevant actions happening on in the file system and I wanted to make sure the record is not added or deleted if those actions fail.

I've marked it as a ToDo for v1.16.